lib/chef/resource/helpers/path_helpers.rb

```diff
@@ -0,0 +1,41 @@
+require "chef-utils/dist" unless defined?(ChefUtils::Dist)
+class Chef
+  module ResourceHelpers
+    # Helpers for path manipulation
+    module PathHelpers
+      extend self
+      # The habitat binary path for Infra Client
+      # @return [String]
+      def chef_client_hab_binary_path
```

Comment:
This should be able to determine the current binary or habitat package directly?